src/core.h

```diff
@@ -2,54 +2,40 @@
 #define CORE_H_
 
 #include <vector>
+#include <list>
 #include <memory>
 #include <random>
 
-class Character {
-  public:
-    double GetMoney();
-    void SetMoney(double money);
-
-  private:
-    double money_;
-};
+#include "action.h"
 
+class Action;
```

Comment:
Having the circular includes is fragile, because depending on the order they are included in a .cc things might not compile. I think it would be better for this file to just have the forward declaration without including action.h.

Reply:
interesting, so just declare that there is a class called Action, without worrying about the interface or structure of that class? I'm trying to wrap my head around the different cases here: core.h needs to know that such a class exists (not it's interface), core.cpp needs to know the interface, action.cpp needs to know the interface/structure (of both stuff in core.h and action.h). does that make sense? is that a common setup? 

Like, what's the dependency graph, what are the nodes and edges in that graph? what's a healthy dependency graph vs an unhealthy one?

Reply:
I think I'm going to refactor this so there's the game state which will just be the static state of the game, not the simulation, and then a decision engine that is running the simulation which will in turn know about actions. So this should turn the dependency graph into a dag (dengine knows about actions, both of them know about cvc). but I think I want to sit on that until I do the bigger action refactor in an upcoming change